lib/src/app_handler.dart

```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+
+// Described an specific application which can open a file
+class AppHandler {
+  String name;
+  String url;
+  Uint8List? icon;
+
+  @internal
+  AppHandler(this.name, this.url, {this.icon});
```

Comment:
I personally prefer to not mix regular parameter with named ones. Flutter's SDK mostly also follows this style.